packages/core/src/hooks/useQueryParams.ts

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import qs from 'qs';
+import { useState } from 'react';
+import { useLocation, useNavigate } from 'react-router-dom';
+import { useDebounce } from 'react-use';
+
+function stringify(queryParams: any): string {
```

Comment:
Can we serialize arrays as multiple entries of the same key?

`filters[Type]=a&filters[Type]=b` instead of `filters[Type][0]=a&filters[Type][1]=b`

Reply:
Think there might be an option in the qs library to do this, it just means less characters in the querystring 👍

Reply:
Yes qs has a lot of options!

I can try that again, but I noticed that most settings of qs have drawbacks and limitations in terms of the types that the resulting object has. That's the reason for the comment in that function 😉

Reply:
A I remember what the difference between `filters[Type]=a&filters[Type]=b` and `filters[Type][0]=a&filters[Type][1]=b` is:

* `filters[Type][0]=a&filters[Type][1]=b` allows to detect that the value is an array, even if only one value is set (e.g. `filters[Type][0]=a`. So we can pass the result of qs directly to the components.
* `filters[Type]=a&filters[Type]=b` doesn't allow to detect an array if only one element is available: `filters[Type]=a`. qs parses it as a simple string, so we need some additional code that allows both a simple string and an array for the tags.

You have similar problems if you choose `,` as a separator. Instead of `[]` it would also be possible to use `.`, that would look better as `.` doesn't have to be escaped as part of an URL, but has similar problems.

@benjdlambert  If the later is desired, I can implement it.

Reply:
Yeah maybe not worth the hassle and good enough for now. If we come up with some other idea in the future I don't consider breaking these params to be that big of an issue tbh, not at this point anyway.